### DISCUSSION TITLE
Extension interface for loco.

### DISCUSSION THREAD

**Discussion by rnbokade** [loco-rs/loco#576] [General]
I come from Django land, where you can create a new supporting library just by creating a new Django app, creating functionality in it and publishing it as a package. These could then be referenced in the settings and Voila! you can start using the application features.
Can we expect similar interface in loco in someway... or even different way of making extensions for loco basically.
For example if I want to create a new authentication strategy and publish it, i should be able to create a package for it and just publish. Similarly, let's say I want to create a package which implements webhooks for some service, let's say stripe... I should somehow be able to declare the models and controllers for it and simply plug it into my other loco.rs app in someway.
In my opinion, this would make a very beautiful addition to boost the ecosystem of extensions for loco.rs

**Comment by jondot** [comment]
I believe you mean embedded apps or similar. Yes your app should end up as a Rust crate like any other so test that and just use it like a library
